Add description of `libs` folder to the `packages` README (#624)
hashintel/hash: packages/README.md

Our [HASH](https://hash.ai/platform/hash) codebase can be found in `hash`. HASH is a data-driven, schema-centric, all-in-one workspace built atop the [Block Protocol](https://blockprotocol.org/).

## `engine`

`engine` houses the codebase of our next-generation version of [hEngine](https://hash.ai/platform/engine) - a versatile agent-based simulation engine written in Rust.

## `blocks`

`blocks` contains the source code for all of HASH's open-source Block Protocol blocks.

## `libs`

> `libs` contains the source code for the software development libraries which HASH has published publicly. These are intended for use within other software projects and are unlikely to provide utility outside of this context. Write-ups of most can be found on the [HASH website for developers](https://hash.dev/).

Reviewer: Why they are unlikely to provide utility outside of this? At least the error library is designed to work on as many projects as possible.

Reviewer 2: I think the risk of someone reading this and not understanding what a software library is are pretty low, so I'd keep it short:
```suggestion
`libs` contains the source code for the software development libraries which HASH has published for general use in building other projects. Write-ups of most can be found on the [HASH website for developers](https://hash.dev/).
```
